lib/jobs/generate-tag.js

```diff
@@ -41,8 +41,13 @@ function generateTagJobFactory(
     function GenerateTagJob(options, context, taskId) {
         GenerateTagJob.super_.call(this, logger, options, context, taskId);
 
-        this.nodeId = context.target || options.nodeId;
-        assert.isMongoId(this.nodeId, 'context.target || options.nodeId');
+        if (options.nodes) {
+            this.nodes = options.nodes;
+            assert.arrayOfString(this.nodes, 'options.nodes');
+        } else {
+            this.nodeId = context.target || options.nodeId;
+            assert.isMongoId(this.nodeId, 'context.target || options.nodeId');
```

Comment:
Whoops, no this warrants a change (these have been mostly updated piecemeal rather than all at once). Good catch.